### ISSUE TITLE
failed compilation in FreeBSD 10.3

### ISSUE THREAD

**Issue by alvespms** [tidyverse/readxl#221]
```
cc -I/usr/local/lib/R/include -DNDEBUG -Iunix -I. -DLIBICONV_PLUG -isystem /usr/local/include -I"/usr/local/lib/R/library/Rcpp/include"   -fpic  -O2 -pipe  -DLIBICONV_PLUG -fstack-protector -isystem /usr/local/include -fno-strict-aliasing  -c xlstool.c -o xlstool.o
xlstool.c:167:12: **error**: static declaration of 'asprintf' follows non-static
      declaration
static int asprintf(char **ret, const char *format, ...)
           ^
/usr/include/stdio.h:399:6: note: previous declaration is here
int      asprintf(char **, const char *, ...) __printflike(2, 3);
         ^
1 error generated.
*** Error code 1
```

For information
> version
               _                           
platform       i386-portbld-freebsd10.1    
arch           i386                        
os             freebsd10.1                 
system         i386, freebsd10.1           
status                                     
major          3                           
minor          3.1                         
year           2016                        
month          06                          
day            21                          
svn rev        70800                       
language       R                           
version.string R version 3.3.1 (2016-06-21)
nickname       Bug in Your Hair

**Comment by hadley** [comment]
cc @jimhester. Any ideas?

**Comment by hadley** [comment]
Maybe this is the same as #189 which is fixed in #215

**Comment by jimhester** [comment]
Yes it looks like that this is the same issue, maybe we just `#define HAVE_APRINTF 1` unconditionally for unix-alikes, I think all the common R architectures should have it.
